### ISSUE TITLE
No access point discovered yet error on ALFA AWUS1900 Monitor Mode Chipset 

### ISSUE THREAD

**Issue by Et0hacker** [wifiphisher/wifiphisher#1177]
Before opening a new issue, please make sure that what you are about to report has not been reported already or is not included on the FAQ page. If not, use the following template for reporting.

Please keep in mind that our developers will prioritize your issue if you provide a detailed and extensive description.

**Version**: Use `git show | grep commit` command and paste here the output.

**Description**: Give a more analytical description here. Let us know how we can reproduce the reported issue. Detail the characteristics of your machine (e.g. Linux distribution and wireless card capabilities) and of the victim devices too (e.g. OS or web browser) if possible. The output of the `iw list` and `iwconfig` commands may also be helpful here.

**Script output**: Paste the output of the tool, including any stack traces.

**Comment by Et0hacker** [comment]
![Capture](https://user-images.githubusercontent.com/50541295/63672788-644f8900-c800-11e9-8b34-8e1ce3117949.PNG)

**Comment by Et0hacker** [comment]
> Wiphy phy0
	max # scan SSIDs: 9
	max scan IEs length: 2304 bytes
	max # sched scan SSIDs: 0
	max # match sets: 0
	max # scan plans: 1
	max scan plan interval: -1
	max scan plan iterations: 0
	Retry short limit: 7
	Retry long limit: 4
	Coverage class: 0 (up to 0m)
	Supported Ciphers:
		* WEP40 (00-0f-ac:1)
		* WEP104 (00-0f-ac:5)
		* TKIP (00-0f-ac:2)
		* CCMP-128 (00-0f-ac:4)
	Available Antennas: TX 0 RX 0
	Supported interface modes:
		 * IBSS
		 * managed
		 * AP
		 * monitor
		 * P2P-client
		 * P2P-GO
	Band 1:
		Capabilities: 0x19f3
			RX LDPC
			HT20/HT40
			Static SM Power Save
			RX Greenfield
			RX HT20 SGI
			RX HT40 SGI
			TX STBC
			RX STBC 1-stream
			Max AMSDU length: 7935 bytes
			DSSS/CCK HT40
		Maximum RX AMPDU length 65535 bytes (exponent: 0x003)
		Minimum RX AMPDU time spacing: 16 usec (0x07)
		HT Max RX data rate: 404 Mbps
		HT TX/RX MCS rate indexes supported: 0-23
		Bitrates (non-HT):
			* 1.0 Mbps
			* 2.0 Mbps
			* 5.5 Mbps
			* 11.0 Mbps
			* 6.0 Mbps
			* 9.0 Mbps
			* 12.0 Mbps
			* 18.0 Mbps
			* 24.0 Mbps
			* 36.0 Mbps
			* 48.0 Mbps
			* 54.0 Mbps
		Frequencies:
			* 2412 MHz [1] (20.0 dBm)
			* 2417 MHz [2] (20.0 dBm)
			* 2422 MHz [3] (20.0 dBm)
			* 2427 MHz [4] (20.0 dBm)
			* 2432 MHz [5] (20.0 dBm)
			* 2437 MHz [6] (20.0 dBm)
			* 2442 MHz [7] (20.0 dBm)
			* 2447 MHz [8] (20.0 dBm)
			* 2452 MHz [9] (20.0 dBm)
			* 2457 MHz [10] (20.0 dBm)
			* 2462 MHz [11] (20.0 dBm)
			* 2467 MHz [12] (20.0 dBm) (no IR)
			* 2472 MHz [13] (20.0 dBm) (no IR)
			* 2484 MHz [14] (20.0 dBm) (no IR)
	Band 2:
		Capabilities: 0x19f3
			RX LDPC
			HT20/HT40
			Static SM Power Save
			RX Greenfield
			RX HT20 SGI
			RX HT40 SGI
			TX STBC
			RX STBC 1-stream
			Max AMSDU length: 7935 bytes
			DSSS/CCK HT40
		Maximum RX AMPDU length 65535 bytes (exponent: 0x003)
		Minimum RX AMPDU time spacing: 16 usec (0x07)
		HT Max RX data rate: 404 Mbps
		HT TX/RX MCS rate indexes supported: 0-23
		VHT Capabilities (0x03c15bb2):
			Max MPDU length: 11454
			Supported Channel Width: neither 160 nor 80+80
			RX LDPC
			short GI (80 MHz)
			TX STBC
			SU Beamformer
			SU Beamformee
			+HTC-VHT
		VHT RX MCS set:
			1 streams: MCS 0-9
			2 streams: MCS 0-9
			3 streams: MCS 0-9
			4 streams: not supported
			5 streams: not supported
			6 streams: not supported
			7 streams: not supported
			8 streams: not supported
		VHT RX highest supported: 1300 Mbps
		VHT TX MCS set:
			1 streams: MCS 0-9
			2 streams: MCS 0-9
			3 streams: MCS 0-9
			4 streams: not supported
			5 streams: not supported
			6 streams: not supported
			7 streams: not supported
			8 streams: not supported
		VHT TX highest supported: 1300 Mbps
		Bitrates (non-HT):
			* 6.0 Mbps
			* 9.0 Mbps
			* 12.0 Mbps
			* 18.0 Mbps
			* 24.0 Mbps
			* 36.0 Mbps
			* 48.0 Mbps
			* 54.0 Mbps
		Frequencies:
			* 5180 MHz [36] (20.0 dBm) (no IR)
			* 5200 MHz [40] (20.0 dBm) (no IR)
			* 5220 MHz [44] (20.0 dBm) (no IR)
			* 5240 MHz [48] (20.0 dBm) (no IR)
			* 5260 MHz [52] (20.0 dBm) (no IR, radar detection)
			* 5280 MHz [56] (20.0 dBm) (no IR, radar detection)
			* 5300 MHz [60] (20.0 dBm) (no IR, radar detection)
			* 5320 MHz [64] (20.0 dBm) (no IR, radar detection)
			* 5500 MHz [100] (20.0 dBm) (no IR, radar detection)
			* 5520 MHz [104] (20.0 dBm) (no IR, radar detection)
			* 5540 MHz [108] (20.0 dBm) (no IR, radar detection)
			* 5560 MHz [112] (20.0 dBm) (no IR, radar detection)
			* 5580 MHz [116] (20.0 dBm) (no IR, radar detection)
			* 5600 MHz [120] (20.0 dBm) (no IR, radar detection)
			* 5620 MHz [124] (20.0 dBm) (no IR, radar detection)
			* 5640 MHz [128] (20.0 dBm) (no IR, radar detection)
			* 5660 MHz [132] (20.0 dBm) (no IR, radar detection)
			* 5680 MHz [136] (20.0 dBm) (no IR, radar detection)
			* 5700 MHz [140] (20.0 dBm) (no IR, radar detection)
			* 5720 MHz [144] (20.0 dBm) (no IR, radar detection)
			* 5745 MHz [149] (20.0 dBm) (no IR)
			* 5765 MHz [153] (20.0 dBm) (no IR)
			* 5785 MHz [157] (20.0 dBm) (no IR)
			* 5805 MHz [161] (20.0 dBm) (no IR)
			* 5825 MHz [165] (20.0 dBm) (no IR)
			* 5845 MHz [169] (disabled)
			* 5865 MHz [173] (disabled)
			* 5885 MHz [177] (disabled)
	Supported commands:
		 * new_interface
		 * set_interface
		 * new_key
		 * start_ap
		 * new_station
		 * set_bss
		 * join_ibss
		 * set_pmksa
		 * del_pmksa
		 * flush_pmksa
		 * remain_on_channel
		 * frame
		 * set_channel
		 * connect
		 * disconnect
	Supported TX frame types:
		 * IBSS: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * managed: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * AP/VLAN: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-client: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
		 * P2P-GO: 0x00 0x10 0x20 0x30 0x40 0x50 0x60 0x70 0x80 0x90 0xa0 0xb0 0xc0 0xd0 0xe0 0xf0
	Supported RX frame types:
		 * IBSS: 0xd0
		 * managed: 0x40 0xd0
		 * AP: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * AP/VLAN: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
		 * P2P-client: 0x40 0xd0
		 * P2P-GO: 0x00 0x20 0x40 0xa0 0xb0 0xc0 0xd0
	WoWLAN support:
		 * wake up on anything (device continues operating normally)
	software interface modes (can always be added):
		 * monitor
	interface combinations are not supported
	Device supports scan flush.
	Supported extended features:

**Comment by Et0hacker** [comment]
Here its creating monitor mode on same wlan0 why it is not creating separate interface for monitor mode.

**Comment by Et0hacker** [comment]
@wifiphisher

**Comment by Et0hacker** [comment]
@raffaele-forte @jaseg @mscherer @d33tah @tgalyean

**Comment by d33tah** [comment]
Pinging @sophron in case you somehow didn't notice this :p

**Comment by Et0hacker** [comment]
@sophron

**Comment by pavonis9** [comment]
@Et0hacker Can you type the **command** you're using.

**Comment by Et0hacker** [comment]
Wifiphisher -nD or wifiphisher

**Comment by pavonis9** [comment]
@Et0hacker So typically you should have two wifi interfaces to work with wifiphisher one wich support AP mode ( As I can see the one that you have now is supporting AP mode ) and the second one to deauth ( You can also use -nD to deauth manually as you have done ).

First don't put your interface in monitor mode just keep it in managed and wifiphisher will take care of the rest.

Try specifying to wifiphisher the interface wich support AP mode by -aI and try also with his built in deauth using the extension interface using -eI.

So the final command will be as :

`wifiphisher -aI wlan1 -eI wlan2 `

If there's still no AP shown try avoiding that step using:

`wifiphisher -aI your-ap-interface -eI your-extension-interface -e WIFI-TEST` 

This will make an AP with the name WIFI-TEST. And we using this command to see if your card have an issue or not.

Try setting your card in monitor mode and start airodump-ng to see if he gives you the Acces points nearby.

And if it didn't work try with : wifiphisher --logging and paste the output here ( You'll find the log as wifiphisher.log in your home directory )

Please answer all my questions to see what's the problem if this haven't solve your issue.

**Comment by Et0hacker** [comment]
2019-09-03 10:41:43,247 -        wifiphisher.pywifiphisher - INFO -
Starting Wifiphisher
2019-09-03 10:41:43,491 -        wifiphisher.common.opmode - INFO -
Starting OP_MODE5 (0x5)
2019-09-03 10:41:43,621 -        wifiphisher.pywifiphisher - INFO -
Selecting wfphshr-wlan0 for deauthentication and wlan0 for the rogue Access
Point
2019-09-03 10:41:43,622 -        wifiphisher.pywifiphisher - INFO -
Unblocking interfaces
2019-09-03 10:41:43,683 -        wifiphisher.pywifiphisher - INFO - Killing
all interfering processes

after this on terminal no accesspoint(wifi) found showing! also see 1 thing
that after i put my card into monitor mode instead of assigning separate
interface for monitor mode its assigning or converting monitor mode in only
on wlan0 not creating separate interface for monitor mode!! i think due to
this wifiphisher is not working
Here is the log
Commands i have given:
wifiphisher --logging ( did this command without putting my card into
monitor mode) and airodump and aireplay are working perfectly!

**Comment by pavonis9** [comment]
It seems that I reproduce the same problem when I keep Network manager running and by default wifiphisher kill's it :

So when I used  `wifiphisher -kN` I had the same situation.

Try to kill network manager once you boot up.

`service NetworkManager stop`

And check it if it still running or not by : 

`service NetworkManager status`

The last thing I can tell you is to try with this : 

`wifiphisher -nD -nE -aI your-main-interface`

Try this command with each interface that you have and typically it should work.

One thing that I have noticed is this : 

> Selecting wfphshr-wlan0 for deauthentication and wlan0 for the rogue Access Point

Can you explain this please.
Is both your interfaces cards are named wlan0 ?
Or you put one in monitor mode then you plug the secondary interface ?
Or you have only one interface card ?

Here is mine take a look : 

` wifiphisher.pywifiphisher - INFO - Selecting wlan0 for deauthentication and wlan1 for the rogue Access Point`

**Comment by Et0hacker** [comment]
Hey,
Thats what i am saying, its having both mode in one interface like after i
enable monitor mode its not creating separate interface its SUCCESFULLY
converts wlan0 to monitor mode but in on only wlan0 not creating interface
like wlan0mon etc

Thanks

**Comment by Et0hacker** [comment]
And i am having only having one card that is alfa awus190

**Comment by pavonis9** [comment]
@Et0hacker Wifiphisher can work with one interface but it's recommended to work with two of them.
When you using only one ( As far as I know ) you can use it only for the AP and no deauth will be there because you can't use the same card for AP and deauth in the same time.

You have to sacrifice the deauth to work with wifiphisher and cross your fingers for someone to be connected to your network. 

Here are some extra details and I hope you read them carefully :

https://github.com/wifiphisher/wifiphisher/issues/652

https://github.com/wifiphisher/wifiphisher/issues/525

**Comment by Et0hacker** [comment]
see this error of roguehostapd...i already installed but still its showing
this error

**Comment by pavonis9** [comment]
**What error ?**

You can use hostapd instead :

`wifiphisher --force-hostapd`

**Comment by Et0hacker** [comment]
successfully started using roguehostapd but serve payload attack not
working also network manager one method is also not working please check
this error i am facing after closing wifiphisher

**Comment by pavonis9** [comment]
>successfully started using roguehostapd but serve payload attack not working
 
Take a look here : 

`/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/data/phishing-pages/plugin_update/html/update`

If this directory exists you need to modify the payload there with your own.

And also change the name of the payoload in payload path of`config.ini` at : 

`/usr/local/lib/python2.7/dist-packages/wifiphisher-1.4-py2.7.egg/wifiphisher/data/phishing-pages/plugin_update/`

> also network manager one method is also not working please check this error i am facing after closing wifiphisher.

What error. Again I'm not able to tell you anything unless you provide me with extra info.

**At least provide me with a screenshot and paste the error output here.**

**Comment by Et0hacker** [comment]
See attachment i provied you sceenshot

**Comment by pavonis9** [comment]
I have noticed that you're responding me from email.
Come and take a look in Github and show the attachment that you're talking about.

**Comment by Et0hacker** [comment]
Ok sir

**Comment by linghushaoxia** [comment]
> `wifiphisher -nD -nE -aI your-main-interface`

I hve the same problem, the command is useful!
wifiphisher -nD -nE -aI your-main-interface
now wifiphisher woks well.
thank you !
